src/vim.h

```diff
@@ -439,8 +439,11 @@ typedef unsigned int u8char_T;	// int is 32 bits or more
  * The vimlong_T has sizeof(vimlong_T) >= 2 * sizeof(int).
  * One use is simple handling of overflow in int calculations.
  */
+#ifdef HAVE_NO_LONG_LONG
```

Comment:
```suggestion
#if defined(VMS) && defined(VAX)
// unsupported compiler
```

I scanned `vim.h` again. Using VMS/VAX looks consistent with the rest of the file. As you mentioned "VAX is the only one".

I agree with your comment (but not your conclusion)

> this clearly explains why we are doing this acrobatics with the types

The answer, particularly in this architectural file, is VAX/VMS. A further comment could be included here. Needing to go to some other file obscures the situation.